src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -46,6 +49,12 @@
     arg_type=name_type,
     options_list=['--domain-topic-name'])
 
+key_name_type = CLIArgumentType(
+    help='Key name to regenerate, which can be either \'key1\' or \'key2\'.',
+    arg_type=name_type,
+    options_list=['--key-name'],
+    completer=get_keys_completion_list)
```

Comment:
A completer isn't really need for this. The simplest way would simply be:

```Python
key_values = ['key1', 'key2']  # this *should* be an enum somewhere in your SDK

with self.argument_context(…) as c:
  c.argument('key_name', arg_type=get_enum_type(key_values))
```

Then you can cut out all of that completer code. Completers are intended for when you need to make REST API calls to get the completions, not for static lists of values.

Reply:
Thanks for the suggestion. Fixed.

FYI, key1 and key2 are not enum values in swagger or sdk 

![image](https://user-images.githubusercontent.com/13698754/59556334-82ce1400-8f75-11e9-96f8-b5a7be9ae711.png)

![image](https://user-images.githubusercontent.com/13698754/59556337-94172080-8f75-11e9-98ff-ee9703ba9654.png)